aioxmpp/presence/service.py

```diff
@@ -236,6 +236,183 @@ def handle_presence(self, st):
                 self.on_changed(st.from_, st)
 
 
+class DirectedPresenceHandle:
+    """
+    Represent a directed presence relationship with a peer.
+
+    Directed Presence is specified in :rfc:`6121` section 4.6. Since the users
+    server is not responsible for distributing presence updates to peers to
+    which the client has sent directed presence, special handling is needed.
```

Comment:
Perhaps we need even more special handling when sending directed presence to peers with from or both subscription (as the RFC specifies that those still receive presence broadcasts). So the current code sends the presence twice to those. (Probably we simply do not want to use this service with those we share subscription with due to the different processing by the server. This should be documented.)

Reply:
We indeed need to document that care should be taken. It’s not as trivial as that, for example when stuff like XEP-0186 (Invisible Command) comes into play.